### ISSUE TITLE
Error rate growth while finetuning

### ISSUE THREAD

**Issue by hforghani** [speechbrain/speechbrain#2808]
### Describe the bug

I finetuned Whisper large-v3 on 2 Persian datasets of my own for example

- dataset 1: 485 h
- dataset 2 : 137 h

up to 241 epochs. I used the recipe CommonVoice/ASR/transformer/train_with_whisper.py with a few customizations for my data.  The charts are below. Train loss, Validation loss, validation WER, and validation CER. (I don't know why validation loss goes up but never mind! it is not my case) Everything is OK!

<img src="https://github.com/user-attachments/assets/9d7a8473-241c-4d38-8080-cd64af24397f" width="500" >

Then I have continued finetuning on another dataset:

- dataset 3: 25 h

**This dataset is more important for me.** and the charts went as below:

<img src="https://github.com/user-attachments/assets/f73eece4-0565-421a-b3b6-58f83cde86fc" width="500" >

In another experiment I finetuned large-v3 just on dataset 3. The result was:

<img src="https://github.com/user-attachments/assets/5a8fd13e-5fc7-42e4-a700-b604bc659895" width="500" >

In another experiment in each 40-size batch I sampled:

- 10 from dataset 1
- 10 from dataset 2
- 20 from dataset 3

The result was:

<img src="https://github.com/user-attachments/assets/e33eb484-23c6-44ca-b31a-792b2d36f601" width="500" >

This is my params for both 1-241 epochs and after 242:

```yaml
freeze_whisper: False
freeze_encoder: True
number_of_epochs: 1000
weight_decay: 0.01
lr_whisper: 1e-5
warmup_steps: 500
max_grad_norm: 2.0
sorting: ascending
precision: bf16 # bf16, fp16 or fp32 (fp16 on gpu)
eval_precision: bf16 # (fp16 on gpu)
sample_rate: 16000
```

I changed lr_whisper and weight_decay as:

```yaml
weight_decay: 0.05
lr_whisper: 1e-6
```
but almost the same result obtained.

What is wrong?

### Expected behaviour

I expect WER and CER go down as 1-241 epochs.

### Relevant Log Output

```shell

```

**Comment by AtlantaANPR** [comment]
Hi, can you please tell me how to fine tune spkrec-ecapa-voxceleb with pretrained weighs on our custom dataset, and also what are the requirements for this.
